include/rapidjson/writer.h

```diff
@@ -46,6 +46,11 @@ RAPIDJSON_DIAG_PUSH
 RAPIDJSON_DIAG_OFF(4127) // conditional expression is constant
 #endif
 
+#ifdef __hpux
+// disable warning unreachable-code
+#pragma diag_suppress 2111
+#endif
```

Comment:
You should use `diag_push`, `diag_pop` to restore the original selection of the application.

Does the HP-UX compiler support something like `_pragma`/`_Pragma`?  In this case, you can add support for the `RAPIDJSON_DIAG_*` macros on this platform as well, see `rapidjson.h`.